fearless/Modules/Staking/Analytics/Stake/AnalyticsStakePresenter.swift

```diff
@@ -0,0 +1,118 @@
+import Foundation
+import BigInt
+import SoraFoundation
+
+final class AnalyticsStakePresenter {
+    weak var view: AnalyticsStakeViewProtocol?
+    let wireframe: AnalyticsStakeWireframeProtocol
+    let interactor: AnalyticsStakeInteractorInputProtocol
+    private let viewModelFactory: AnalyticsStakeViewModelFactoryProtocol
+    private let localizationManager: LocalizationManagerProtocol
+    private let logger: LoggerProtocol?
+
+    private var rewardsData: [SubqueryStakeChangeData]?
+    private var selectedPeriod = AnalyticsPeriod.default
+    private var selectedPeriodDiff = 0
```

Comment:
Consider to store Calendar and base date here. Also subscribe to day change and update the interface